src/value.rs

```diff
@@ -0,0 +1,1613 @@
+        let mut endianity = constants::DW_END_default;
+        let mut attrs = entry.attrs();
+        while let Some(attr) = attrs.next()? {
+            match attr.name() {
+                constants::DW_AT_byte_size => byte_size = attr.udata_value(),
+                constants::DW_AT_encoding => {
+                    if let AttributeValue::Encoding(x) = attr.value() {
+                        encoding = Some(x);
+                    }
+                }
+                constants::DW_AT_endianity => {
+                    if let AttributeValue::Endianity(x) = attr.value() {
+                        endianity = x;
+                    }
+                }
+                _ => {}
+            }
+        }
+
+        if endianity != constants::DW_END_default {
+            // TODO: we could check if it matches the reader endianity,
+            // but normally it would use DW_END_default in that case.
+            return Ok(None);
+        }
+
+        if let (Some(encoding), Some(byte_size)) = (encoding, byte_size) {
+            Ok(ValueType::from_encoding(encoding, byte_size))
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+impl Value {
+    /// Return the `ValueType` corresponding to this `Value`.
+    pub fn value_type(&self) -> ValueType {
+        match *self {
+            Value::Generic(_) => ValueType::Generic,
+            Value::I8(_) => ValueType::I8,
+            Value::U8(_) => ValueType::U8,
+            Value::I16(_) => ValueType::I16,
+            Value::U16(_) => ValueType::U16,
+            Value::I32(_) => ValueType::I32,
+            Value::U32(_) => ValueType::U32,
+            Value::I64(_) => ValueType::I64,
+            Value::U64(_) => ValueType::U64,
+            Value::F32(_) => ValueType::F32,
+            Value::F64(_) => ValueType::F64,
+        }
+    }
+
+    /// Read a `Value` with the given `value_type` from a `Reader`.
+    pub fn parse<R: Reader>(value_type: ValueType, mut bytes: R) -> Result<Value> {
+        let value = match value_type {
+            ValueType::I8 => Value::I8(bytes.read_i8()?),
+            ValueType::U8 => Value::U8(bytes.read_u8()?),
+            ValueType::I16 => Value::I16(bytes.read_i16()?),
+            ValueType::U16 => Value::U16(bytes.read_u16()?),
+            ValueType::I32 => Value::I32(bytes.read_i32()?),
+            ValueType::U32 => Value::U32(bytes.read_u32()?),
+            ValueType::I64 => Value::I64(bytes.read_i64()?),
+            ValueType::U64 => Value::U64(bytes.read_u64()?),
+            ValueType::F32 => Value::F32(bytes.read_f32()?),
+            ValueType::F64 => Value::F64(bytes.read_f64()?),
+            _ => return Err(Error::UnsupportedTypeOperation),
+        };
+        Ok(value)
+    }
+
+    /// Convert a `Value` to a `u64`.
+    ///
+    /// The `ValueType` of `self` must be integral.
+    /// Values are sign extended if the source value is signed.
+    pub fn to_u64(self, addr_mask: u64) -> Result<u64> {
```

Comment:
We should file a follow up issue to implement `TryFrom<Value> for u64` in a month and a half, when it hits stable.

Reply:
I don't think that allows the `addr_mask` parameter.